test/OpenTelemetry.Instrumentation.Dependencies.Tests/HttpWebRequestTests.Basic.net461.cs

```diff
@@ -62,9 +62,12 @@ public void Dispose()
         [Fact]
         public async Task HttpDependenciesInstrumentationInjectsHeadersAsync()
         {
-            var spanProcessor = new Mock<SpanProcessor>();
-            var tracer = TracerFactory.Create(b => b.AddProcessorPipeline(p => p.AddProcessor(_ => spanProcessor.Object)))
-                .GetTracer(null);
+            var activityProcessor = new Mock<ActivityProcessor>();
+            using var shutdownSignal = OpenTelemetrySdk.EnableOpenTelemetry(b =>
+            {
+                b.SetProcessorPipeline(c => c.AddProcessor(ap => activityProcessor.Object));
```

Comment:
@cijothomas This build-up pattern was really confusing. It looks like internally ActivityProcessor is intended to be chained but there is nothing in the abstract class that enforces it. We never set a "Next" or make sure that the chain is called. Probably need to do some more work in this area?

Reply:
Its the same as what we have with TracerBuilder. SpanProcessors are chained, but its not enforced in base class.
The OpentelemetryBuilder is very basic at this stage (doesn't have all the things as TracerBuilder). But if the current "next" pattern is confusing, we can address it as well. Would you need it for this PR or we can handle it separate ? 
I can add a 1e here: https://github.com/open-telemetry/opentelemetry-dotnet/issues/684

Reply:
Separate PR sounds good to me. ASP.NET Core Middleware uses a similar pattern:

https://github.com/dotnet/aspnetcore/blob/425c196cba530b161b120a57af8f1dd513b96f67/src/Http/Http.Abstractions/src/IMiddleware.cs#L19
https://github.com/dotnet/aspnetcore/blob/0e38046bce72ac0a916a08763a7fb3e79c786749/src/Http/Http/src/Builder/ApplicationBuilder.cs#L82

Reply:
Yes. We used same in Azure Application Insights as well.